developer-extension/src/panel/hooks/useEvents.ts

```diff
@@ -15,24 +15,35 @@ export interface EventFilters {
   query: string
 }
 
-export function useEvents() {
+export function useEvents(preserveEvents: boolean) {
   const [events, setEvents] = useState<StoredEvent[]>([])
   const [filters, setFilters] = useState<EventFilters>({
     sdk: ['rum', 'logs'],
     query: '',
   })
 
-  useEffect(
-    () =>
-      listenRequests((newEvents) => {
-        setEvents((oldEvents) =>
-          [...newEvents, ...oldEvents]
-            .sort((first: any, second: any) => second.date - first.date)
-            .slice(0, MAXIMUM_LOGGED_EVENTS)
-        )
-      }),
-    []
-  )
+  useEffect(() => {
+    const removeListener = listenRequests((newEvents) => {
+      setEvents((oldEvents) =>
+        [...newEvents, ...oldEvents]
+          .sort((first: any, second: any) => second.date - first.date)
+          .slice(0, MAXIMUM_LOGGED_EVENTS)
+      )
+    })
+
+    if (!preserveEvents) {
+      const clearCurrentEvents = (details: chrome.webNavigation.WebNavigationTransitionCallbackDetails) => {
+        if (details.transitionType === 'reload') setEvents([])
+      }
+      chrome.webNavigation.onCommitted.addListener(clearCurrentEvents)
+      return () => {
+        removeListener()
+        chrome.webNavigation.onCommitted.removeListener(clearCurrentEvents)
+      }
+    }
```

Comment:
Because of this early return, the `removeListener` won't always be called. What about using two `useEffect` like I suggested [here](https://github.com/DataDog/browser-sdk/pull/1825#discussion_r1029152982)?